compiler/damlc/tests/src/DA/Test/Packaging.hs

```diff
@@ -253,6 +253,26 @@ tests damlc = testGroup "Packaging"
         checkDarFile darFiles "B" "C.hi"
         checkDarFile darFiles "B" "C.hie"
 
+    , testCase "Dalf dependencies get package id suffices" $ withTempDir $ \projDir -> do
+        createDirectoryIfMissing True (projDir </> "daml")
+        writeFileUTF8 (projDir </> "daml/A.daml") $ unlines
+          [ "daml 1.2"
+          , "module A where"
+          , "data A = A ()"
+          ]
+        writeFileUTF8 (projDir </> "daml.yaml") $ unlines
+          [ "sdk-version: " <> sdkVersion
+          , "name: proj"
+          , "version: 0.1.0"
+          , "source: daml"
+          , "dependencies: [daml-prim, daml-stdlib]"
+          ]
+        buildProject projDir
+        let dar = projDir </> ".daml/dist/proj-0.1.0.dar"
+        assertBool "proj-0.1.0.dar was not created." =<< doesFileExist dar
+        darFiles <- Zip.filesInArchive . Zip.toArchive <$> BSL.readFile dar
+        checkNotDarFile darFiles "." "daml-prim.dalf"
```

Comment:
This looks a bit convoluted, why don’t we assert that a file with the expected naming scheme, e.g., something of the form `daml-prim-*.dalf` exists instead of checking that `daml-prim.dalf` does not exist?